stlab/concurrency/utility.hpp

```diff
@@ -58,57 +58,75 @@ future<T> make_exceptional_future(std::exception_ptr error, E executor) {
 
 template <typename T>
 T blocking_get(future<T> x) {
-    T result;
-    std::exception_ptr error;
+    boost::optional<T> result;
+    std::exception_ptr error = nullptr;
 
-    bool set{false};
+    bool flag{0};
```

Comment:
Why do you set the flag to zero and not to false?

Reply:
Fixed. (had changed it to an int when trying to figure out the bug... forgot to change the initializer back).